draft-ietf-privacypass-batched-tokens.md

```diff
@@ -319,13 +327,116 @@ struct {
 If the FinalizeBatch function fails, the Client aborts the protocol. Token
 verification works exactly as specified in {{ISSUANCE}}.
 
+# Arbitrary Batched Verifiable Token
+
+## Client-to-Issuer Request {#arbitrary-client-to-issuer-request}
+
+The Client first creates all TokenRequest it wants to batch. To do so, the client follows
+protocol describing issuance, such as {{ISSUANCE, Section 5.1}} or {{ISSUANCE, Section 6.1}}.
+
+The Client then creates a BatchedTokenRequest structured as follows:
+
+~~~tls
+struct {
+  uint16_t token_type;
```

Comment:
[RFC 9576](https://www.rfc-editor.org/rfc/rfc9576#section-3.5-2) specifies that
> The structure and semantics of the TokenRequest and TokenResponse messages depend on the issuance protocol and token type being used

This means that while all existing TokenRequest start with `token_type`, it's a convention and might break in future Issuance protocols.

Therefore the restriction to one token type per batched request, instead of prepending token_type to all TokenRequest. This also seems like a way to fail fast for an issuer, which would know if they can support the batched requests, knowing both the number of request and their type.

Reply:
That applies to the application/private-token-request media type. But if we use a different media type, as we've done here, that restriction no longer applies.

Reply:
Would restricting this new issuance protocol to TokenRequest that use `{ uint16_t token_type, ... }` be a possibility? It limits the future TokenRequest, if they want to use Arbitrary Batching.

Reply:
the proposal has been updated to reflect the comment just above.